src/psyclone/domain/lfric/kernel/inter_grid_arg.py

```diff
@@ -78,12 +80,20 @@ def create_from_fparser2(fparser2_tree):
         :returns: an instance of InterGridArg.
         :rtype: :py:class:`psyclone.domain.lfric.kernel.InterGridArg`
 
+        raises ValueError: if the metadata is not in the correct form.
+
         '''
         InterGridArg.check_fparser2(
             fparser2_tree, nargs=5, encoding=Fortran2003.Structure_Constructor)
+        InterGridArg.check_first_arg(fparser2_tree, "InterGrid")
         datatype, access, function_space = \
-            InterGridArg.get_type_access_and_fs(fparser2_tree)
+            InterGridArg.get_type_access_and_fs(
```

Comment:
Again, should this be an instance method? Actually, I see that at this point we haven't yet constructed an instance of `InterGridArg`. Could you access these properties in the method using `cls.access_arg_index` perhaps?

Reply:
Made into a class method. This makes it much nicer, thanks. I also made the `get_and_check_function_space()` method into a class method for the same reason.